editor/src/document/document_message_handler.rs

```diff
@@ -43,6 +45,52 @@ impl DocumentsMessageHandler {
 	pub fn active_document_mut(&mut self) -> &mut DocumentMessageHandler {
 		&mut self.documents[self.active_document_index]
 	}
+	fn generate_new_document_name(&self) -> String {
+		let digits = ('0'..='9').collect::<Vec<char>>();
+		let mut doc_title_numbers = self
+			.documents
+			.iter()
+			.map(|d| {
+				if d.name.starts_with(DEFAULT_DOCUMENT_NAME) && d.name.ends_with(digits.as_slice()) {
+					let (_, number) = d.name.split_at(DEFAULT_DOCUMENT_NAME.len());
+					number.trim().parse::<usize>().unwrap()
+				} else {
+					1
+				}
```

Comment:
I think this code might actually break the logic, we should not assume a one when the other criteria don't match,
only if the document title matches but not the ends with slice, otherwise they should be ignored